### PR TITLE
Automatic build and release

### PR DESCRIPTION
**Pull request by ChristianFeldmann**
There are 3 changes in this that I would like to propose:
 - 1. There was a bit of (It think) unconsistent behavior when building shared libraries. On windows, the `dll` file would go into the `bin` output folder while on linux/mac the `.so` and `.dylib` file would go into the `lib` folder. This is a Cmake quirk where dll and so/dylib are handled differently (see https://stackoverflow.com/questions/56514533/how-to-specify-the-output-directory-of-a-given-dll). I think this should be consistent and all of them should go into the runtime folder (`bin`). This also helps with point 2:
 - 2. I added an upload that just attaches the whole runtime folder (`bin`) to the build as an artifact. Then for each build you can just download the compiled files and ... well do whatever you want with them. I also enabled building of static and non static on most platforms. For the ubuntu 16 and 18 this could also be enabled. 
 - 3. I added a release upload of the libraries for windows/mac/ubuntu.20.04. This means if you publish a release with this, these files will be automatically attached to the release. These are the libraries that can be used in YUView. That is why I only attached these. Of course also the executables could be attached to the release here. Up to you guys if you want that.

### CODE REVIEW

File: .github/workflows/Build.yml [hunk 1]
```diff
@@ -13,26 +13,63 @@ jobs:
...
       if: matrix.os != 'windows-2019'
     - name: Build Windows
       run: |
         mkdir build
         cd build
         call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat"
-        cmake .. -DCMAKE_BUILD_TYPE=Release
-        msbuild vvdec.sln
+        cmake ${{matrix.CMAKE_ADD}} ..
+        msbuild /property:Configuration=Release vvdec.sln
```
Comment (ChristianFeldmann) [inline .github/workflows/Build.yml:58]: This was actually a bug. This is how you build the release version on windows. Before, the debug version was built.

Comment (1480c1) [reply .github/workflows/Build.yml:58]: Perhaps it would be better to use `cmake --build . --config Release` so you don't have to use msbuild directly?

Comment (ChristianFeldmann) [reply .github/workflows/Build.yml]: Very good idea. Then you could even switch to a different build system like Ninja ... But for a project this small that is not really necessary.

**Comment by K-os** [comment]
Hi Christian,
Thank you very much for your contribution.

Unfortunately, there are two separate issues concerning this PR:

1. We would like to keep the `lib/bin` separation in the build tree. I think this is not a quirk by CMake, but it just adheres to the platform conventions. On *nix-systems you typically install the binaries in `/usr/{local/}bin/` and libraries in `/usr/{local/}lib/`. I think it would lead to more confusion, if the build step puts the shared libraries into the `bin` folder during build, but during install they end up in `lib`.

2. We have not come to a final decision if we are going to provide official binaries for VVdeC.

So, I would like to put your PR on hold for now, until we have decided on the binaries issue, but feel free to provide the binaries from your repository.

**Comment by ChristianFeldmann** [comment]
Got it. Just wanted to show how this stuff works. Then I will use it in my fork for the releases.

**Pull request closed without merging**